state-sync/storage-service/server/src/lib.rs

```diff
@@ -705,15 +704,23 @@ fn log_storage_response(storage_response: &Result<StorageServiceResponse, Storag
                 sample!(
                     SampleRate::Duration(Duration::from_secs(SUMMARY_LOG_FREQUENCY_SECS)),
                     {
+                        let response = format!("{}", storage_response);
                         debug!(LogSchema::new(LogEntry::SentStorageResponse).response(&response));
                     }
                 );
             } else {
-                debug!(LogSchema::new(LogEntry::SentStorageResponse).response(&response));
+                sample!(
+                    SampleRate::Duration(Duration::from_secs(SUMMARY_LOG_FREQUENCY_SECS)),
```

Comment:
A couple small issues:
1. Storage server summaries are expected to be requested fairly frequently but don't change that drastically, hence the sampling above. This is not true of the rest of the storage server responses, they will all be different and depend on how the client is operating and what the client wants. I'm not sure it makes sense to only log this infrequently. Let's either log it, or don't.
2. If you want to log it infrequently (not sure how helpful it will be) create a new constant. `SUMMARY_LOG_FREQUENCY_SECS` is for the storage server summaries and not all responses. Also, why not wrap the entire `if` statement in a `sample!` instead?

Reply:
Good call - removing the sampling for this for now as this is not actually consuming any significant amount of CPU.